### DISCUSSION TITLE
Build rpm and deb packages

### DISCUSSION THREAD

**Discussion by hakong** [netbox-community/netbox#6344] [General]
### NetBox version

n/a

### Feature type

New functionality

### Proposed functionality

Build RPM and DEB packages for netbox. Anyone discussed this? Would this be possible?

### Use case

Installation, upgrades, and removal on rpm and deb based systems.

**Comment by hakong** [comment]
The netbox package for EPEL 8, is it official or 3rd party? I'm assuming it's a third-party build.
https://src.fedoraproject.org/rpms/netbox

**Comment by tijuca** [comment]
I've started to package NetBox for Debian for some time and have a basically working instance.

NetBox is requiring quite recent Python libraries. As Debian is preparing the next stable release right now getting updated versions of all required libraries into the archive can take some time.

Updates are required for:

- [django-cors-headers](https://tracker.debian.org/pkg/django-cors-headers) 2.2.0 vs 3.7.0
- [django-taggit](https://tracker.debian.org/pkg/django-taggit) 0.24.0 vs. 1.4.0
- [tablib](https://tracker.debianhttps://tracker.debian.org/pkg/python-tablib.org/pkg/python-tablib) 0.130 vs. 3.0.0

This requires some new packages as they are build and/or install dependencies:

- django-cacheops requires funcy
- tablib requires MarkuPy

I also identified some not yet packaged required libraries, these are so far:

- django-cacheops
- django-pglocks
- django-rq
- drf-yasg

The last one is a bit difficult as I'm impossible to rebuild every thing from scratch, this make this packages currently non-free for Debian. And this would need NetBox has to go into the contrib part.

And some problems related to drf-yasg I've discovered also, calling the API definition interface (http[s]://netbox.domain/api/docs/) is leading to an error that needs to get discovered and figured out. Currently I just see an error message about an internal error has occurred.

Also NetBox is requiring [Django version 3.2.x](https://github.com/netbox-community/netbox/blob/develop/requirements.txt#L1) regarding the requirements.txt file. The Debian Django team decided to only [support](https://salsa.debian.org/python-team/packages/python-django/-/blob/debian/sid/debian/README.Django-packaging-policy) the latest LTS version in unstable/testing and by this also in stable. For now this shouldn't be real issue, but in a long term NetBox shouldn't require a Django development version to run on.

Currently I'm in a state to have a working example and being far away from getting this all into Debian soon. Especially the handling of the postinst procedure needs further improvements. I'm not an experienced NetBox user nor have really deeper knowledge about Django internals, bur I'm interested to get this in a clear state. Using docker images isn't a real option on my day job so I'm interested to have NetBox packaged for and in Debian.

**Reply by jeremystretch** [reply]
> For now this shouldn't be real issue, but in a long term NetBox shouldn't require a Django development version to run on.

To clarify, it doesn't, and never has. Production (non-beta) NetBox releases always ship with a stable version of Django. The only difference between a stable Django release and an LTS release is that the LTS release will continue to receive bug/security fixes for a longer time.

Django LTS releases come out only about once every two years; these are the x.2 releases within each major version. [This chart](https://jefftriplett.com/django-release-cycle/) provides a good overview of past and planned releases.

**Comment by tijuca** [comment]
> To clarify, it doesn't, and never has. Production (non-beta) NetBox releases always ship with a stable version of Django. 

That's good to hear. I was just following the data from the [requirements.txt](https://github.com/netbox-community/netbox/blob/master/requirements.txt) file within the last release and there is stated that Django 3.2.4 is needed.
I'm aware about the Django version that is intended to become a LTS version. For now Django 3,2.x hasn't reached unstable/testing as the current version 2.2.x is the targeted version for next stable release which will get released in a few weeks..

So it's more clear to me how I can proceed. Thanks for your feedback!
